producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+
+==== Map Type
+Avro assumes that every key is of a string type and automatically converts every key to a
+string representation. See link:http://avro.apache.org/docs/1.10.0/api/java/org/apache/avro/util/Utf8.html[Utf8] - all keys are converted to this class.
+For example, a map with Integer values will have such Avro schema:
+[source,json]
+----
+{
+  "type":"map",
+  "values":{
+    "type":"map",
+    "values":"int"
+  }
+}
+----
+Please note that there is no field that represents the value for keys in a map.
+
+==== List and Set Types
+Both lists and sets are converted to the Avro array type.
+For example, the resulting type for set and list of an integer type will be:
+[source,json]
+----
+{
+ "type":"array",
+ "items":"int"
+}
+----
+
+==== User Defined Type (UDT) Schema
+The UserDefined is a Record type in Avro.
+For example, such an UserDefined type:
+
+[source,java]
+----
+LinkedHashMap<String, CQLType> udtColumns = new LinkedHashMap<>();
+udtColumns.put("udtcol_1", Native.INT);
+udtColumns.put("udtcol_2", Native.TEXT);
+UserDefined userDefinedType = new UserDefined("ks", "typeName", udtColumns);
+----
+
+will have the following schema:
+
+[source,json]
+----
+{
+"type":"record",
+"name":"typeName",
+"fields":[
+   {
+      "name":"udtcol_1",
+      "type":"int"
+   },
+   {
+      "name":"udtcol_2",
+      "type":"string"
+   }
+ ]
+}
+----
+
+The generated schema also supports nested UserDefined types.
+
+==== Tuple Type
+
+The Tuple is a Record type in Avro. For example, such a Tuple type:
+
+[source,java]
+----
+new Tuple(Native.INT, new Collection(Kind.LIST, Native.TEXT));
+----
+
+will have the following schema:
+
+[source,json]
+----
+{
+"type":"record",
+"name":"tuple_int_list_text__",
+"fields":[
+   {
+      "name":"t_0",
+      "type":"int"
+   },
+   {
+      "name":"t_1",
+      "type":{
+         "type":"array",
+         "items":"string"
+      }
+   }
+ ]
+}
+----
+
+Please note that the name of the record is transformed according to `CqlToAvroTypeConverter#tupleToRecordName(Tuple)` method.
+Every element in the tuple has a name according to `CqlToAvroTypeConverter#toTupleFieldName(int)`.
+The generated schema also supports nested Tuple types.
+
+==== Custom Type
+The custom type is saved as bytes without an attempt to deserialize it. It's the client
+responsibility to deserialize it correctly.
+Currently, the class name form `Custom#getClassName()` is not propagated in the avro message.
+
+== New Topic Creation
+When the mutation event for a topic that does not exists in Kafka arrives, the producer will create the topic automatically if
+link:https://kafka.apache.org/documentation/#allow.auto.create.topics[allow.auto.create.topics] on the Kafka broker is enabled.
+It is enabled on default. It would be best if you were careful because Kafka will automatically create the topic with the default replication factor and number of partitions.
```

Comment:
```suggestion
It is enabled by default. **NOTE**: you should be careful with this setting since Kafka will automatically create the topic with the default replication factor and number of partitions.
```